+++ gc.c
@@ -39,9 +39,6 @@
 
 #define rb_setjmp(env) RUBY_SETJMP(env)
 #define rb_jmp_buf rb_jmpbuf_t
-#ifdef __CYGWIN__
-int _setjmp(), _longjmp();
-#endif
 
 /* Make alloca work the best possible way.  */
 #ifdef __GNUC__
+++ eval.c
@@ -217,9 +217,6 @@
 #  define ruby_setjmp(extra_save, env) \
      ((extra_save), RUBY_SETJMP(env))
 #  define ruby_longjmp(env,val) RUBY_LONGJMP(env,val)
-#  ifdef __CYGWIN__
-int _setjmp(), _longjmp();
-#  endif
 #endif
 
 #include <sys/types.h>
